libgadget/exchange.c

```diff
@@ -362,8 +362,8 @@ domain_build_plan(int (*layoutfunc)(int p), ExchangePlan * plan)
 
     int rank;
     for(rank = 1; rank < NTask; rank ++) {
-        plan->toGoOffset[rank] = plan->toGoSum;
-        plan->toGetOffset[rank] = plan->toGetSum;
+        memcpy(&plan->toGoOffset[rank], &plan->toGoSum, sizeof(plan->toGoSum));
```

Comment:
Could you add a comment here to say that direct assignment breaks compilers like icc?